python/cudf/cudf/core/dataframe.py

```diff
@@ -5708,9 +5707,15 @@ def stack(self, level=-1, dropna=True):
             }
         )
 
-        data_col = libcudf.reshape.interleave_columns(homogenized)
+        result = Series._from_data(
```

Comment:
Can we change the above `homogenized` dataframe construction to go via `DataFrame._from_data`? It would help avoid a lot of index checks and handlings. Like:

```python
homogenized = DataFrame._from_data(
            {
                c: (
                    self._data[c].astype(common_type)
                    if not np.issubdtype(self._data[c].dtype, common_type)
                    else self._data[c].copy()
                )
                for c in self._data
            }
        )
```

Reply:
Done, also for `new_index` code above.